query-planner-js/src/__tests__/buildPlan.interfaceObject.test.ts

```diff
@@ -696,3 +696,151 @@ it('handles @interfaceObject in nested entity', () => {
     }
   `);
 });
+
+it('handles @interfaceObject input rewrites when cloning dependency graph', () => {
+  const subgraph1 = {
+    name: 'S1',
+    typeDefs: gql`
+      type Query {
+        i: I!
+      }
+
+      interface I @key(fields: "i1") {
+        i1: String!
+        i2: T
+      }
+
+      type T @key(fields: "t1", resolvable: false) {
+        t1: String!
+      }
+
+      type U implements I @key(fields: "i1") {
+        id: ID!
+        i1: String!
+        i2: T @shareable
+      }
+    `,
+  };
+
+  const subgraph2 = {
+    name: 'S2',
+    typeDefs: gql`
+      type I @interfaceObject @key(fields: "i1") {
+        i1: String!
+        i2: T @shareable
+        i3: Int
+      }
+
+      type T @key(fields: "t1", resolvable: false) {
+        t1: String!
+      }
+    `,
+  };
+
+  const subgraph3 = {
+    name: 'S3',
+    typeDefs: gql`
+      type T @key(fields: "t1") {
+        t1: String!
+        t2: String! @shareable
+        t3: Int
+      }
+    `,
+  };
+
+  const subgraph4 = {
+    name: 'S4',
+    typeDefs: gql`
+      type T @key(fields: "t1") {
+        t1: String!
+        t2: String! @shareable
+        t4: Int
+      }
+    `,
+  };
+
+  const [api, queryPlanner] = composeAndCreatePlanner(
+    subgraph1,
+    subgraph2,
+    subgraph3,
+    subgraph4,
+  );
```

Comment:
in order to trigger input rewrites we need to federate the call to `@interfaceObject` (S1 -> S2) and then in order to trigger dependency graph copy we need to have multiple query plan options (i.e. S2 -> S3 or S2 -> S4).